### PR TITLE
[Enhancement]Use latest tag for bentoml containerize command

### PR DESCRIPTION
**Pull request by yubozhao**
## Description

Current behavior:
If a user provides a docker image tag value without `:` for `bentoml containerize` command, BentoML will use the bento's version instead for the tag
```
$ bentoml containerize irisclassifier:version123 -t my_custom_image
> Docker image ->  my_custom_image:version123
```

This PR will add an option `--tag-latest/--do-not-tag-latest` to `bentoml containerize` command

```
$ bentoml containerize irisclassifier:latest
> Docker image ->  irisclassifier:actual_version,   irisclassifier:latest

$ bentoml containerize irisclassifier:latest --tag my_image
> Docker image-> my_image:actual_version, my_image:latest

$ bentoml containerize irisclassifier:latest --do-not-tag-latest
> Docker image -> irisclassifier:actual_version
```

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New feature and improvements (non-breaking change which adds/improves functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code Refactoring (internal change which is not user facing)
- [ ] Documentation
- [ ] Test, CI, or build

## Component(s) if applicable
- [ ] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [ ] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [x] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/cli/bento_service.py [hunk 1]
```diff
@@ -353,11 +354,16 @@ def install_completion(append, shell, path):
     @click.option(
         '-p', '--password', type=click.STRING, required=False,
     )
-    def containerize(bento, push, tag, build_arg, username, password):
+    @click.option(
+        '--tag-latest/--do-not-tag-latest',
+        default=True,
+        help='BentoML will create a special latest tag for the image.',
```
Comment (jackyzha0) [inline bentoml/cli/bento_service.py:360]:
```suggestion
        help='BentoML will add the special 'latest' tag to the built image.',
```

File: bentoml/cli/bento_service.py [hunk 2]
```diff
@@ -418,6 +424,21 @@ def containerize(bento, push, tag, build_arg, username, password):
                     )
                 ):
                     _echo(line)
+
+            if tag_latest:
+                image_name, current_tag = tag.split(":")
+                if current_tag == "latest":
+                    _echo('Skipping building "latest" tag.')
```
Comment (jackyzha0) [inline bentoml/cli/bento_service.py:431]:
```suggestion
                    _echo('Already has 'latest' tag, skipping adding 'latest' tag.')
```

File: bentoml/cli/bento_service.py [hunk 3]
```diff
@@ -418,6 +424,21 @@ def containerize(bento, push, tag, build_arg, username, password):
...
+                else:
+                    _echo(f'Tagging latest: {image_name}:latest')
```
Comment (jackyzha0) [inline bentoml/cli/bento_service.py:433]:
```suggestion
                    _echo(f'Adding 'latest' tag: {image_name}:latest')
```

File: bentoml/cli/bento_service.py [hunk 4]
```diff
@@ -418,6 +424,21 @@ def containerize(bento, push, tag, build_arg, username, password):
...
+                    try:
+                        docker_api.tag(image=tag, repository=image_name, tag='latest')
+                        _echo(f'Finished tagging {image_name}:lateset')
```
Comment (jackyzha0) [inline bentoml/cli/bento_service.py:436]:
```suggestion
                        _echo(f'Added tag {image_name}:latest')
```

File: bentoml/cli/bento_service.py [hunk 5]
```diff
@@ -418,6 +424,21 @@ def containerize(bento, push, tag, build_arg, username, password):
...
+                    except docker.errors.APIError as error:
+                        _echo(
+                            f'Failed to tag "latest" for {tag}: {error}',
```
Comment (jackyzha0) [inline bentoml/cli/bento_service.py:439]:
```suggestion
                            f'Failed to add tag 'latest' to {tag}: {error}',
```

**Review by jackyzha0 — COMMENTED** [review]
Some small spelling and phrasing things! Also, should we be testing this addition?

**Comment by yubozhao** [comment]
`containerize` function will be part of the Yatai server in the upcoming versions, will look back at this issue again.

It will handle multiple tag options.
```
bentoml containerize IrisClassifier:latest --tag mydeployment:version --tag mydeployment:latest --tag other-tag
```

**Pull request closed without merging**